proposals/1.0.0/packedMod.md

```diff
@@ -0,0 +1,167 @@
+
+__Implementaion:__ Required
+
+__Minimal requirements & dependencies:__ R0010
+
+__Methods:__
+
+* Get the path of the packed mod by using drag and drop to interpret or install it.
+* Search automatic or manual (in an specific location) for ccmod files. Use the path to interpret or install the mod.
+* Manually input the path to a ccmod file.
+* Manually input the URL to a ccmod file.
+
+#### Interpreter
+
+This describes how packed mods should be interpreted. This can be used to show additional information, change the mod, publish it into a databse or interprete the mod for other purposes. This method will not install the mod.
+
+1. Extraction the zipped data and store it in memory
+2. Validate package.json
+
+What the tool is doing with the mod afterwards is highly sepecific. This also can be excracting the zipped data and store them on HDD (+R0030).
+
+__Examples:__
+
+```
+$ cli show path/to/mod.ccmod
+
+CrossCode mod
+modtitle v 1.0.0
+moddescription
+
+Required
+- CrossCode 1.0.0
+- CLS 1
+
+Dependencies
+- x
+
+[...]
+```
+
+```
+$ cli show path/to/mod.ccmod -json
+{ [...] }
+```
+
+* Click on [Load mod] in an tool and input the path for example.ccmod to show and edit mod information.
+* Click on [Load mod] in an tool and extract the data to a specific path on HDD (+R0030).
+
+#### Installation
+
+This describes how packed mods should be installed by a modloader.
+
+1. Extraction the zipped data and store it on HDD [OR] in memory
+2. __Optional:__ Validate package.json
+3. __Optional:__ Ask to user if the mod should be installed and show it's information (requires 2)
+4. Installation of the mod if accepted
+
+__Examples:__
+
+* The modloader is seaching for ccmod files in a defined mods folder
+* The path to a ccmod file is put into a form to add the mod to the game
+
+
+### File association
+
+The tool is setting up the system to load ccmod files with a tool by adding the tool to the file association list. This allows the user to install or interpert packed mods with a simple doubleclick on the packed mod.
+
+__Implementaion:__ Optional
+
+__Minimal requirements & dependencies:__ R0010, R0020, R0030, R0040, R0050
+
+__Methods:__
+
+* Load ccmod files with a doubleclick
+
+#### Setup
+
+Defines the setup of the ccmod file association.
+
+1. Check if a program is already associated with ccmod
+
+Possiblities if a program is associated:
+
+* Skip the setup
+* Ask the user if the tool should be associated instead (only once) and if register it if accepted
+* Do not register the tool if declined
+
+Possiblities if no program is associated:
+
+* Set the file association automatic
+* Ask the user if the tool should be associated instead (only once) and if register it if accepted
+* Do not register the tool if declined
+
+#### Interpreter
```

Comment:
How is interpreting meant with file association? Will the game be started with the mod running? Should it add the mod to a running game?